Fixed Prerequisites for LAB and GIR classes

liuric opened the pull request.

liuric committed:
Modified search to not modify search query for LAB subjects and use the rawValue field instead of the toString method
Additionally, modified regex to accept singular prereqs

liuric committed:
Removed Debug prints

venkatesh-sivaraman commented on app/src/main/java/com/base12innovations/android/fireroad/models/course/CourseSearchEngine.java:
```diff
@@ -172,7 +172,7 @@ public static void exactMatch(EnumSet<Filter> filter) {
                 for (String comp : queryComps) {
                     boolean found = false;
-                    String regex = filters.contains(Filter.MATCHES) ? "^.*[^A-z0-9]" + Pattern.quote(comp) + "[^A-z0-9].*$" : "";
+                    String regex = filters.contains(Filter.MATCHES) ? "^.*" + Pattern.quote(comp) + ".*$" : "";
```
Is this change necessary? It should be responsible for doing an exact (full-word) match as opposed to "contains".

liuric replied:
I see, it just happens that the field sometimes just has the word/course itself (just "6.009") so the character requirement rejects it. It should be fixed now (added * to the original regex), but I got a weird bug where the last search result went offscreen and added a margin to fix it

venkatesh-sivaraman replied:
Makes perfect sense, I like the new fix.

venkatesh-sivaraman reviewed (commented):
Looks great, just had one question before you merge.

liuric committed:
Re-added full-word matching
Added margin to prevent the bottom end of the search courses recyclerview from being offscreen, which prevented user interaction with the last item when there are > 9 items

venkatesh-sivaraman reviewed (approved):
Looks ready to merge!
